### PR TITLE
fix(MeshMultiZoneService): handle protocol with routes

### PR DESCRIPTION
**Pull request by michaelbeaumont**
We weren't properly computing the conf and handling the ports as HTTP

I won't close the linked issue yet but after this PR, policies applied to `MeshMultiZoneService` should just work after loosening validation. Tested with https://github.com/kumahq/kuma/pull/11060

### Checklist prior to review

- [x] [Link to relevant issue][1] as well as docs and UI issues -- Part of #11042 
- [x] This will not break child repos: it doesn't hardcode values (.e.g "kumahq" as a image registry) and it will work on Windows, system specific functions like `syscall.Mkfifo` have equivalent implementation on the other OS --
- [x] Tests (Unit test, E2E tests, manual test on universal and k8s) --
  - Don't forget `ci/` labels to run additional/fewer tests
- [x] Do you need to update [`UPGRADE.md`](../blob/master/UPGRADE.md)? --
- [x] Does it need to be backported according to the [backporting policy](../blob/master/CONTRIBUTING.md#backporting)? ([this](https://github.com/kumahq/kuma/actions/workflows/auto-backport.yaml) GH action will add "backport" label based on these [file globs](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L6), if you want to prevent it from adding the "backport" label use [no-backport-autolabel](https://github.com/kumahq/kuma/blob/master/.github/workflows/auto-backport.yaml#L8) label) --

> Changelog: skip

[1]: https://docs.github.com/en/issues/tracking-your-work-with-issues/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword

### CODE REVIEW

File: pkg/plugins/policies/core/rules/resourcerules_test.go [hunk 1]
```diff
@@ -307,6 +308,44 @@ var _ = Describe("Compute", func() {
 		Expect(rule.Conf).To(Equal([]interface{}{"conf-2"}))
 	})
 
+	It("should return Mesh rule if MeshMultiZoneService is not found", func() {
```
Comment (slonka) [inline pkg/plugins/policies/core/rules/resourcerules_test.go:311]: Why is this related?

**Review by slonka — COMMENTED** [review]
Please help me understand why is this mesh fallback needed 🤔

File: pkg/plugins/policies/core/rules/resourcerules_test.go (hunk 1 above)
Comment (michaelbeaumont) [reply pkg/plugins/policies/core/rules/resourcerules_test.go:311]: Because `Compute` wasn't working. It's just the fact that a `kind: Mesh` rule applies if there's no `kind: MeshMultiZoneService`

**Review by slonka — APPROVED** [review]
